framework/db/ActiveRecordInterface.php

```diff
@@ -427,4 +427,19 @@ public function unlink($name, $model, $delete = false);
      * @return mixed the database connection used by this AR class.
      */
     public static function getDb();
+
+    /**
+     * Creates an active record instance.
+     *
+     * This method is called internally by framework for instantiating new record object.
+     * It is not meant to be used for creating new records directly.
+     *
+     * You may override this method if the instance being created
+     * depends on the row data to be populated into the record.
+     * For example, by creating a record based on the value of a column,
+     * you may implement the so-called single-table inheritance mapping.
+     * @param array $row row data to be populated into the record (can be empty).
+     * @return ActiveRecordInterface the newly created active record
+     */
+    public static function instantiate($row = []);
```

Comment:
Need to have this one as is and, additionally, add another method w/o params that is creating an instance but isn't filling it. It would be used for cases when we're getting metadata such as attr. labels or joinWith relation metadata.

In 2.1 we may call the one w/o arguments `instantiate()` and the one w/ `$row` hydrate.